File: backend/waste-service/src/main/java/com/ecobridge/waste_service/service/WasteService.java
package com.ecobridge.waste_service.service;

import com.ecobridge.waste_service.dto.request.CreateWasteRequest;
import com.ecobridge.waste_service.dto.request.UpdateWasteRequest;
import com.ecobridge.waste_service.dto.response.RecyclerDashboardResponse;
import com.ecobridge.waste_service.dto.response.WasteResponse;
import com.ecobridge.waste_service.dto.response.WasteStatsResponse;


import java.util.List;
import java.util.UUID;

public interface WasteService {

    WasteResponse createWaste(CreateWasteRequest request);

    WasteResponse getWasteById(UUID id);

    List<WasteResponse> getMyWaste();

    WasteResponse updateWaste(UUID id, UpdateWasteRequest request);

    void deleteWaste(UUID id);

    List<WasteResponse> getAllWaste();

    WasteStatsResponse getWasteStats();
    WasteResponse reserveWaste(UUID wasteId);

    List<WasteResponse> getAvailableWaste();

    List<WasteResponse> getMyPickups();

    WasteResponse completePickup(UUID wasteId);

    List<WasteResponse> getPickupHistory();

    RecyclerDashboardResponse getRecyclerDashboard();
}

File: frontend/ecobridge-frontend/src/services/wasteService.js
import api from "../api/axios";

export const getMyWaste = () => {
    return api.get("/api/v1/waste/my");
};

export const createWaste = (data) => {
    return api.post("/api/v1/waste", data);
};

export const updateWaste = (id, data) => {
    return api.put(`/api/v1/waste/${id}`, data);
};

export const deleteWaste = (id) => {
    return api.delete(`/api/v1/waste/${id}`);
};

export const getWasteById = (id) => {
    return api.get(`/api/v1/waste/${id}`);
};

export const getAvailableWaste = () => {

    return api.get(
        "/api/v1/waste/available"
    );

};

export const reserveWaste = (id) => {

    return api.put(
        `/api/v1/waste/${id}/reserve`
    );

};

export const getMyPickups = () => {

    return api.get(
        "/api/v1/waste/my-pickups"
    );

};

export const completePickup = (id) => {

    return api.put(
        `/api/v1/waste/${id}/complete`
    );

};

export const getPickupHistory = () => {

    return api.get(
        "/api/v1/waste/pickup-history"
    );

};

export const getRecyclerDashboard = () => {

    return api.get(
        "/api/v1/waste/recycler-dashboard"
    );

};